src/Symfony/Component/Security/Http/RememberMe/TokenBasedRememberMeServices.php

```diff
@@ -61,6 +62,10 @@ protected function processAutoLoginCookie(array $cookieParts, Request $request)
             throw new \RuntimeException(sprintf('The UserProviderInterface implementation must return an instance of UserInterface, but returned "%s".', get_debug_type($user)));
         }
 
+        if (!$user instanceof PasswordAuthenticatedUserInterface) {
+            throw new \RuntimeException(sprintf('Class "%s" must implement "%s" for using "%s".', get_debug_type($user), PasswordAuthenticatedUserInterface::class, __CLASS__));
+        }
```

Comment:
I don't like this: remember shouldn't depend on passwords imho (e.g. when using login links or webauthn)

Reply:
Oh, nvm this class is going to be removed anyways (it's the legacy remember me service)